service/src/test/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,209 @@
+                new BloodPressureMapper(
+                        messageContext, randomIdGeneratorService, new StructuredObservationValueMapper(),
+                        codeableConceptCdMapper, new ParticipantMapper()),
+                new ConditionLinkSetMapper(
+                        messageContext, randomIdGeneratorService, codeableConceptCdMapper, participantMapper),
+                new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new DocumentReferenceToNarrativeStatementMapper(messageContext),
+                new ImmunizationObservationStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new MedicationStatementMapper(messageContext, codeableConceptCdMapper, participantMapper, randomIdGeneratorService),
+                new ObservationToNarrativeStatementMapper(messageContext, participantMapper),
+                new ObservationStatementMapper(
+                        messageContext,
+                        new StructuredObservationValueMapper(),
+                        new PertinentInformationObservationValueMapper(),
+                        codeableConceptCdMapper,
+                        participantMapper
+                ),
+                new RequestStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new DiagnosticReportMapper(messageContext, specimenMapper, participantMapper, randomIdGeneratorService)
+        );
+
+        final EncounterMapper encounterMapper = new EncounterMapper(messageContext, encounterComponentsMapper);
+
+        final NonConsultationResourceMapper nonConsultationResourceMapper =
+                new NonConsultationResourceMapper(messageContext, randomIdGeneratorService, encounterComponentsMapper);
+
+        OrganizationToAgentMapper organizationToAgentMapper = new OrganizationToAgentMapper(messageContext);
+        PractitionerAgentPersonMapper practitionerAgentPersonMapper
+                = new PractitionerAgentPersonMapper(messageContext, organizationToAgentMapper);
+
+        final AgentDirectoryMapper agentDirectoryMapper = new AgentDirectoryMapper(practitionerAgentPersonMapper,
+                organizationToAgentMapper);
+
+
+        EhrExtractMapper ehrExtractMapper = new EhrExtractMapper(randomIdGeneratorService, timestampService, encounterMapper,
+                nonConsultationResourceMapper, agentDirectoryMapper, messageContext);
+
+
+        final EhrExtractTemplateParameters ehrExtractTemplateParameters =
+                ehrExtractMapper.mapBundleToEhrFhirExtractParams(getGpcStructuredTaskDefinition, bundle);
+
+        final String ehrExtractContent = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        final String hl7TranslatedResponse = outputMessageWrapperMapper.map(getGpcStructuredTaskDefinition, ehrExtractContent);
+
```

Comment:
messageContext.resetMessageContext() should be executed here otherwise in case there will be more files to parse the ids and will shared between them and it will produce wrong output.